src/libraries/System.Net.Security/src/System/Net/Security/SslAuthenticationOptions.cs

```diff
@@ -49,7 +49,10 @@ internal void UpdateOptions(SslClientAuthenticationOptions sslClientAuthenticati
             IsServer = false;
             RemoteCertRequired = true;
             // RFC 6066 section 3 says to exclude trailing dot from fully qualified DNS hostname
-            TargetHost = sslClientAuthenticationOptions.TargetHost.TrimEnd('.');
+            if (sslClientAuthenticationOptions.TargetHost != null)
```

Comment:
There's an assert on L20 that this is not null.

Reply:
updated and I added new test for this case.